helix-term/src/commands.rs

```diff
@@ -3830,7 +3861,20 @@ pub mod insert {
             .clone()
             .transform(|range| movement::move_prev_word_start(text, range, count));
         doc.set_selection(view.id, selection);
-        delete_selection(cx)
+        delete_selection_insert_mode(cx)
+    }
+
+    pub fn delete_word_forward(cx: &mut Context) {
+        let count = cx.count();
+        let (view, doc) = current!(cx.editor);
+        let text = doc.text().slice(..);
+
+        let selection = doc
+            .selection(view.id)
+            .clone()
+            .transform(|range| movement::move_next_word_start(text, range, count));
+        doc.set_selection(view.id, selection);
+        delete_selection_insert_mode(cx)
```

Comment:
Yeah you still need to build a selection, but rather than use `set_selection` you can just use it directly:
```suggestion
        let selection = doc
            .selection(view.id)
            .clone()
            .transform(|range| movement::move_next_word_start(text, range, count));
        // then delete
        let transaction = Transaction::change_by_selection(doc.text(), selection, |range| {
            (range.from(), range.to(), None)
        });
        doc.apply(&transaction, view_id);
```

You can just inline the `delete_selection_insert_mode` code.

Reply:
Thanks for the suggestions, updated the code.